### ISSUE TITLE
Go gRPC stream last log missing request

### ISSUE THREAD

**Issue by yolocs** [abcxyz/lumberjack#360]
### TL;DR

https://github.com/abcxyz/lumberjack/blob/main/clients/go/pkg/audit/interceptor.go#L238-L251

We stage the last request in the stream in https://github.com/abcxyz/lumberjack/blob/main/clients/go/pkg/audit/interceptor.go#L288

On happy path, in the next response in the stream we will log the last request https://github.com/abcxyz/lumberjack/blob/main/clients/go/pkg/audit/interceptor.go#L350-L359
But on error, we will get a `handleErr` instead https://github.com/abcxyz/lumberjack/blob/main/clients/go/pkg/audit/interceptor.go#L244-L251 and we never check the last req. As a result, on error the last log will miss the last request.

### Environment Details

```markdown
N/A
```
